### DISCUSSION TITLE
Automatic image updates with generic git repositories

### DISCUSSION THREAD

**Discussion by sagikazarmark** [fluxcd/flux2#2507] [Q&A]
I'm trying to bootstrap a repository with a Bitbucket repository. Unfortunately, it's not supported natively, so I have to use it as a generic git repository.

I'd like to support automatic image updates though, so I created a separate bitbucket user (deploy keys in bitbucket are readonly) with write access. I added the generated SSH public key to that user.

At this point, I'm not entirely sure what's going to happen. The [documentation](https://fluxcd.io/docs/guides/image-update/) and [other sources](https://github.com/fluxcd/flux2/discussions/1819) suggest that I should run the bootstrap command with a `--read-write-key` option...but there is no such option for the bootstrap command or the generic git bootstrap command.

Is the generic git repo going to work with automatic image updates? Will it be able to write back to the repository without explicitly telling it it's a read-write access?

**Comment by stefanprodan** [comment]
When using `flux bootstrap git` it’s up to you to set the deploy key and its permissions.

**Comment by kingdonb** [comment]
It does seem like the docs are pretty heavily leaning into `--read-write-key` when that's only available on a subset of platforms.

You can add a deploy key manually, as Stefan suggested, but we might need to revisit our docs to make sure this is clearer. For example:

> GitLab and other Git platforms
> You can install Flux and bootstrap repositories hosted on GitLab, BitBucket, Azure DevOps and any other Git provider that support SSH or token-based authentication. When using SSH, make sure the deploy key is configured with write access --read-write-key. Please see the [installation guide](https://fluxcd.io/docs/installation/) for more details.

This section of the docs doesn't really clarify that you can add the read-write key yourself, or explain at all how to do that.
